addons/binding/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/discovery/NanoleafMDNSDiscoveryParticipant.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.nanoleaf.internal.discovery;
+
+import static org.openhab.binding.nanoleaf.internal.NanoleafBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import javax.jmdns.ServiceInfo;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.mdns.MDNSDiscoveryParticipant;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.nanoleaf.internal.NanoleafBindingConstants;
+import org.openhab.binding.nanoleaf.internal.NanoleafHandlerFactory;
+import org.openhab.binding.nanoleaf.internal.OpenAPIUtils;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link NanoleafMDNSDiscoveryParticipant} is responsible for discovering new Nanoleaf controllers (bridges).
+ *
+ * @author Martin Raepple - Initial contribution
+ */
+
+@Component(immediate = true, configurationPid = "discovery.nanoleaf")
+@NonNullByDefault
+public class NanoleafMDNSDiscoveryParticipant implements MDNSDiscoveryParticipant {
+
+    // see http://forum.nanoleaf.me/docs/openapi#_gf9l5guxt8r0
+    private static final String SERVICE_TYPE = "_nanoleafapi._tcp.local.";
```

Comment:
Maybe move SERVICE_TYPE to the Constants class? In case it should ever be adapted, it would help discovering it.